src/providers/Wallet.js

```diff
@@ -165,4 +83,75 @@ function WalletProvider({ children }) {
   )
 }
 
+function useConnection() {
+  const wallet = useWallet()
+  const { account, connector } = wallet
+  /* We need  to pass down on the providers tree a preferred network in case that there is no network connnected
+  or the connected network is not supported in order to show some data and also to react to the network drop down selector changes */
+  const [preferredNetwork, setPreferredNetwork] = useState(getPreferredChain())
+  const [switchingNetworks, setSwitchingNetworks] = useState(false)
+
+  const connect = useCallback(
```

Comment:
Removed the `addEthereumChain` function call. The rationale being that we will only need to add a new chain if the user selects one of the chains in the dropdown of the garden list filters. In that case the user will be prompted to switch networks and the `switchNetwork` function already handles the network addition if it doesn´t exist.